packages/langium/test/grammar/langium-grammar-validator.test.ts

```diff
@@ -30,4 +30,51 @@ describe('Langium grammar validation', () => {
             property: {name: 'terminal'}
         });
     });
+
+    // verifies that interfaces can't extend type unions, especially inferred ones
+    test('Interfaces extend only interfaces', async () => {
+        const validationResult = await validate(`
+        grammar G
+
+        // generates an inferred union of 2 other inferred interfaces
+        entry InferredUnion: InferredI1 | InferredI2;
+
+        // for the case where an inferred interface extends another inferred interface
+        InferredI0: InferredI1;
+
+        // just for the sake of generating a pair of inferred interfaces
+        InferredI1: 'InferredI1' InferredI1=ID;
+        InferredI2: 'InferredI2' InferredI2=ID;
+
+        // should fail...
+        interface DeclaredExtendsUnion extends InferredUnion {}
+
+        // we can extend an inferred type that has no parent
+        interface DeclaredExtendsInferred1 extends InferredI2 {}
+
+        // we can extend an inferred type that extends other inferred types
```

Comment:
Comments could be the name of the new test.

Reply:
Comments have to explain, why something is what it is. I think we should form could so that it is easy to read without checking comments… Comments can outdate, very fast.